File: LeetCode/Easy/0392-is-subsequence/0392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int i = 0, j = 0;
        int n = t.length(), m = s.length();
        char[] sCharArray = s.toCharArray();
        char[] tCharArray = t.toCharArray();

        if (m < 1) return true;

        while (i < n) {
            if (tCharArray[i] == sCharArray[j]) {
                j++;
            }
            i++;
            if (j == m) return true;
        }
        return false;
    }
}

File: LeetCode/Easy/0392-is-subsequence/0392-is-subsequence.kt
class Solution {
    fun isSubsequence(s: String, t: String): Boolean {
        if (s.isEmpty()) {
            return true
        } else if (t.isEmpty()) {
            return false
        }
        var sIndex = 0
        var tIndex = 0
        while (tIndex < t.length && sIndex < s.length) {
            if (s[sIndex] == t[tIndex++]) {
                sIndex++
            }
        }
        return sIndex == s.length
    }
}